### DISCUSSION TITLE
docs: machine.RTC

### DISCUSSION THREAD

**Discussion by big-vl** [micropython/micropython#13049] [ESP32] [answered]
Please tell me if there is a difference in using Timer and irq in machine.RTC which is currently missing in esp32.

**Comment by peterhinch** [comment]
I have moved this to `discussions/ESP32` where it will be more visible than in the MicroPython bug tracker.

Please explain in more detail what you are trying to do. The `machine.Timer` class is quite separate from `machine.RTC`.

**Reply by big-vl** [reply]
At the moment I am using this implementation to perform the change over time
        # TIMER REFRESH CURENT TIMER
        t1 = Timer(1)
        t1.init(period=30000, mode=Timer.PERIODIC, callback=lambda b: self.check_clock_change())

**Reply by big-vl** [reply]
I use a timer to display new time on displays, I need to know when the time has changed and display it on the display, but the method does not have alarm and irq, in the version MicroPython v1.20.0-dirty on 2023-09-09; ESP32 module (spiram) with ESP32

Could you tell me if there is currently an implementation of irq so that I can build a version with a callback to update the time.

**Reply by karfas** [reply]
Please **read** the *fine* documentation on
- https://docs.micropython.org/en/latest/library/machine.Timer.html
- https://docs.micropython.org/en/latest/library/machine.RTC.html

The timer provided by the `Timer` class *is* interrupt-driven. 
I have no idea why you need "irq" and "alarm". 
The RTC alarm has the additional feature to wake up from different sleep states (and might run even when the ESP32 is in deep-sleep).

**Reply by big-vl** [reply]
ok, using a loop to update the time is not the right solution, resources are unlimited. Besides these hours, I already have cycles. Or implement a design pattern to use the update in one cycle, but that would be just as sad, until the update is completed, other functions will not be executed, send it to the scheduler as a solution only.

**Reply by karfas** [reply, marked as answer]
Nothing in micropython will magically run faster or in the background. A function scheduled using `micropython.schedule()` will block execution of everything else.
To achieve background processing you will need to use an extra thread or [asyncio](https://github.com/peterhinch/micropython-async/blob/master/v3/docs/TUTORIAL.md).

**Reply by big-vl** [reply]
For some reason it seemed to me that interruptions were executed in a separate thread
